api/v1alpha4/azurecluster_default.go

```diff
@@ -104,22 +100,44 @@ func (c *AzureCluster) setSubnetDefaults() {
 	}
 	setSecurityRuleDefaults(&cpSubnet.SecurityGroup)
 
-	if nodeSubnet.Name == "" {
-		nodeSubnet.Name = generateNodeSubnetName(c.ObjectMeta.Name)
-	}
-	if len(nodeSubnet.CIDRBlocks) == 0 {
-		nodeSubnet.CIDRBlocks = []string{DefaultNodeSubnetCIDR}
-	}
-	if nodeSubnet.SecurityGroup.Name == "" {
-		nodeSubnet.SecurityGroup.Name = generateNodeSecurityGroupName(c.ObjectMeta.Name)
-	}
-	setSecurityRuleDefaults(&nodeSubnet.SecurityGroup)
-	if nodeSubnet.RouteTable.Name == "" {
-		nodeSubnet.RouteTable.Name = generateNodeRouteTableName(c.ObjectMeta.Name)
+	c.Spec.NetworkSpec.UpdateControlPlaneSubnet(cpSubnet)
+
+	var nodeSubnetFound bool
+	var nodeSubnetCounter int
+	for i := range c.Spec.NetworkSpec.Subnets {
+		if c.Spec.NetworkSpec.Subnets[i].Role == SubnetNode {
+			nodeSubnetCounter++
+			nodeSubnetFound = true
+			if c.Spec.NetworkSpec.Subnets[i].Name == "" {
+				c.Spec.NetworkSpec.Subnets[i].Name = withIndex(generateNodeSubnetName(c.ObjectMeta.Name), i)
```

Comment:
They are not exactly the same though. We want the subnet CIDR to start in 1, but I'm starting subnet names with 0. That's why I use a different index.

Reply:
I suggested that because there is a chance that we might skip a number if there is control plane subnet in between. Eg. If the 2nd element is cp subnet, we'll have subnet-0, ~subnet-1 because it's not a node subnet~, subnet-2.

Reply:
ok, let's use `nodeSubnetCounter`, I think it makes sense.